barretenberg/cpp/src/barretenberg/api/api_client_ivc.cpp

```diff
@@ -210,6 +212,46 @@ void ClientIVCAPI::write_solidity_verifier([[maybe_unused]] const Flags& flags,
     throw_or_abort("API function contract not implemented");
 }
 
+bool ClientIVCAPI::check_ivc(const std::filesystem::path& input_path)
+
+{
+    init_bn254_crs(1 << CONST_PG_LOG_N);
+    init_grumpkin_crs(1 << CONST_ECCVM_LOG_N);
+    PrivateExecutionSteps steps;
+    steps.parse(PrivateExecutionStepRaw::load_and_decompress(input_path));
+
+    for (auto [program, precomputed_vk, function_name] :
+         zip_view(steps.folding_stack, steps.precomputed_vks, steps.function_names)) {
+        if (precomputed_vk == nullptr) {
+            info("FAIL: Expected precomputed vk for function ", function_name);
+            return false;
+        }
+        std::shared_ptr<ClientIVC::DeciderProvingKey> proving_key = get_acir_program_decider_proving_key(program);
+        auto computed_vk = std::make_shared<ClientIVC::MegaVerificationKey>(proving_key->proving_key);
+        if (computed_vk->circuit_size != precomputed_vk->circuit_size ||
+            computed_vk->num_public_inputs != precomputed_vk->num_public_inputs ||
+            computed_vk->pub_inputs_offset != precomputed_vk->pub_inputs_offset) {
```

Comment:
for now you could add `pairing_inputs_public_input_key` and `databus_propagation_data` here